unifi/Dockerfile

```diff
@@ -26,78 +26,26 @@
 # 	set-inform http://ip_of_docker_host:8080/inform
 #
 
-FROM ubuntu:16.04
+FROM ubuntu:20.04
 
 # environment settings
 ENV DEBIAN_FRONTEND="noninteractive"
 
-# install deps
-RUN apt-get update && apt-get install -y \
-	ca-certificates \
-	dirmngr \
-	gnupg \
-	--no-install-recommends \
-	&& rm -rf /var/lib/apt/lists/*
-
-# install gosu
-ENV GOSU_VERSION 1.12
-RUN set -ex; \
-	\
-	fetchDeps=' \
-		wget \
-	'; \
-	apt-get update; \
-	apt-get install -y --no-install-recommends $fetchDeps; \
-	rm -rf /var/lib/apt/lists/*; \
-	\
-	dpkgArch="$(dpkg --print-architecture | awk -F- '{ print $NF }')"; \
-	wget -O /usr/local/bin/gosu "https://github.com/tianon/gosu/releases/download/$GOSU_VERSION/gosu-$dpkgArch"; \
-	wget -O /usr/local/bin/gosu.asc "https://github.com/tianon/gosu/releases/download/$GOSU_VERSION/gosu-$dpkgArch.asc"; \
-	\
-# verify the signature
-	export GNUPGHOME="$(mktemp -d)"; \
-    for server in $(shuf -e ha.pool.sks-keyservers.net \
-                            hkp://p80.pool.sks-keyservers.net:80 \
-                            keyserver.ubuntu.com \
-                            hkp://keyserver.ubuntu.com:80 \
-                            pgp.mit.edu) ; do \
-        gpg --keyserver "$server" --recv-keys B42F6819007F00F88E364FD4036A9C25BF357DD4 && break || : ; \
-    done && \
-	gpg --batch --verify /usr/local/bin/gosu.asc /usr/local/bin/gosu; \
-	rm -r "$GNUPGHOME" /usr/local/bin/gosu.asc; \
-	\
-	chmod +x /usr/local/bin/gosu; \
-# verify that the binary works
-	gosu nobody true; \
-	\
-	apt-get purge -y --auto-remove $fetchDeps
-
-# add mongo repo
-RUN apt-key adv --keyserver hkp://keyserver.ubuntu.com:80 --recv 0C49F3730359A14518585931BC711F9BA15703C6 \
-	&& echo "deb [ arch=amd64,arm64 ] http://repo.mongodb.org/apt/ubuntu xenial/mongodb-org/3.4 multiverse" >> /etc/apt/sources.list.d/mongo.list
-
-
-# install packages
-RUN apt-get update && apt-get install -y \
-	binutils \
-	jsvc \
-	mongodb-org-server \
-	openjdk-8-jre-headless \
-	--no-install-recommends \
-	&& rm -rf /var/lib/apt/lists/*
-
 # unifi version
 # From: https://www.ubnt.com/download/unifi/
-ENV UNIFI_VERSION "5.12.72"
+ENV UNIFI_VERSION "7.0.25-43e7fc6711"
 
-# install unifi
+# install unifi and deps
 RUN apt-get update && apt-get install -y \
+		ca-certificates \
+		openjdk-8-jre-headless \
+		gosu \
 		curl \
 		--no-install-recommends \
-	&& rm -rf /var/lib/apt/lists/* \
 	&& curl -o /tmp/unifi.deb -L "https://dl.ubnt.com/unifi/${UNIFI_VERSION}/unifi_sysvinit_all.deb" \
```

Comment:
You should create another layer in the docker image here (line 45).
If the packages exist in the container and they have not changed, why repeat the steps again? It will help in rebuilds in case the unifi deb fails to download.

Reply:
I initially had that, but it means re-running the `apt-get update` step or leaving the `/var/lib/apt/lists/*` in the layer between the runs. The way I have it structured it allows `unifi.deb` to also install any declared dependencies.